Upgrade to ESP8266@2.5.2 broke toolchain for AARCH64 for docker

**Operating environment/Installation (Hass.io/Docker/pip/etc.):** Docker/Hass.io
**ESP (ESP32/ESP8266, Board/Sonoff):** ESP8266

**Description of problem:**

ESPHome recently updated the default arduino framework for ESP8266 to 2.5.2

That version of arduino framework comes with a new xtensa toolchain to get some libc changes (mainly I think exceptions enabled, see also https://github.com/earlephilhower/esp-quick-toolchain).

ESPHome's aarch64 docker images all have a custom toolchain that I compiled some time ago that works with native aarch64 machines (not just armhf that support aarch64, see also https://github.com/esphome/esphome-docker-base/blob/master/gen.py#L9-L22)

That toolchain needs to be rebuilt with the newest toolchain and added to the docker image. Problem is: building the toolchain on a non-native machine takes *ages* (many hours), and the build often happens to break for me (too little disk space etc etc).

Until then, users with the aarch64 version of ESPHome must use `arduino_version: 2.4.2` in their `esphome:` config.

Error message:

```
/data/test8266/.pioenvs/test8266/libFrameworkArduino.a(core_esp8266_main.cpp.o):(.iram.text./root/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_main.cpp.245.5+0x0): undefined reference to `app_entry_custom()'
/data/test8266/.pioenvs/test8266/libFrameworkArduino.a(core_esp8266_main.cpp.o): In function `app_entry':
core_esp8266_main.cpp:(.iram.text./root/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/core_esp8266_main.cpp.245.5+0xa): undefined reference to `app_entry_custom()'
```

<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -3,7 +3,7 @@
 variables:
   DOCKER_DRIVER: overlay2
   DOCKER_HOST: tcp://docker:2375/
-  BASE_VERSION: '2.0.0'
+  BASE_VERSION: '2.0.1'
   TZ: UTC
 
 stages:

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -1,4 +1,4 @@
-ARG BUILD_FROM=esphome/esphome-base-amd64:2.0.0
+ARG BUILD_FROM=esphome/esphome-base-amd64:2.0.1
 FROM ${BUILD_FROM}
 
 COPY . .

--- a/docker/Dockerfile.lint
+++ b/docker/Dockerfile.lint
@@ -1,4 +1,4 @@
-FROM esphome/esphome-base-amd64:2.0.0
+FROM esphome/esphome-base-amd64:2.0.1
 
 RUN \
     apt-get update \

